### PR TITLE
asserting minimum age of containers and images before cleaning up. 

### LINKED ISSUE #40: protect "newest" images from garbage collection
As a follow up on #39 a certain set of new images (e.g. not older than 12 hours) should never be deleted - even if desired free disc space constraints are not fulfilled.

This prevents images from being deleted while they are still in use.

### PR DESCRIPTION
**Pull request by tfoote**
Fixes #40.

This also adds a --min-days  argument and a --dry-run argument to the command line interface.

### CODE REVIEW

**Comment by tfoote** [comment]
This also fixes #43

File: master/jenkins_files/files/home/jenkins-slave/cleanup_docker_images.py [hunk 1]
```diff
@@ -74,14 +77,32 @@ def run_image_cleanup(args):
...
-            logging.info("successfully removed image: %s" % i)
+            if docker_id_older(i, minimum_age):
+                logging.info("removing image %s" % i)
+                if args.dry_run:
+                    logging.info("Dry run >> I would have removed image: %s" % i)
+                else:
+                    remove_docker_image(i)
+                    logging.info("successfully removed image: %s" % i)
+            else:
+                logging.info("skipped removal of image due to age: % s " % i)
```
Comment (dirk-thomas) [inline master/jenkins_files/files/home/jenkins-slave/cleanup_docker_images.py]: Spaces between % and s as well as a taking space.

File: master/jenkins_files/files/home/jenkins-slave/cleanup_docker_images.py [hunk 2]
```diff
@@ -117,8 +144,16 @@ def run_container_cleanup():
                         help='What mount point to introspect')
     parser.add_argument('--logfile', type=str, default='/var/log/jenkins-slave/cleanup_docker_images.log',
                         help='Where to log output')
+    parser.add_argument('--min-days', type=int, default=1,
```
Comment (dirk-thomas) [inline master/jenkins_files/files/home/jenkins-slave/cleanup_docker_images.py]: Wouldn't it be better to allow finer granularity? At least hours.

File: master/jenkins_files/files/home/jenkins-slave/cleanup_docker_images.py [hunk 3]
```diff
@@ -117,8 +144,16 @@ def run_container_cleanup():
...
+                        help='The minimum age of items to clean up.')
+    parser.add_argument('--dry-run', '-n', default=False,
+                    action='store_true',
+                    help='Do not actually clean up, just ')
```
Comment (dirk-thomas) [inline master/jenkins_files/files/home/jenkins-slave/cleanup_docker_images.py]: Incomplete sentence?

**Comment by dirk-thomas** [comment]
Some lines vary from PEP8.

**Comment by dirk-thomas** [comment]
Is there no more efficient way to determine the age without invoking a subprocess for each image / container?

**Comment by tfoote** [comment]
Pep 8 cleaned up. I don't know of a better way to get the age. I can do command line parsing of the output of `docker ps -a` and `docker images -a` but that's going to be very fragile.

**Pull request merged**